### ISSUE TITLE
Feature request: consistent HTTP Proxy support in Lattice

### ISSUE THREAD

**Issue by menicosia** [cloudfoundry-incubator/lattice#181]
In GH Issue #147, @dhilipkumars brought up an interesting point that while we've done some work to enable users behind an HTTP proxy, support is inconsistent.

He wrote:
> @menicosia Thanks again was not able to get lattice working in a new environment primarily because of proxy problems which prevents from downloading any sort of artifact from within the garden-linux Container. :-( Which is why i thought i will try these new commands to run an app locally. That did not work as well because it needs buildpacks to be downloaded. :-(
> 
> buildpacks are downloaded by the container?
> 
> one feedback if we are using VAGRANT_HTTP_PROXY and VAGRANT_HTTPS_PROXY variables to vagrant up lattice then we must also use VAGRANT_NO_PROXY otherwise condencer-bucket.
> 
> below curl would fail.
> 
> curl -s -H 'Content-Type: application/json' -XPOST http://localhost:8980/riak-cs/user --data '{"email":"admin@lattice.cf", "name":"admin_user"}'

When using the [proxy configuration](https://github.com/cloudfoundry-incubator/lattice#proxy-configuration) instructions, the cluster installer sets those directives throughout the deployment - any component that respects those environment variables will use the proxy as configured. Unfortunately, this also includes the components that talk to Lattice's internal blob store.

So that's a bug.

After some discussion, we propose to solve this for Vagrant only, in this way:
```
$ http_proxy=xxx https_proxy=yyy vagrant up
```

This works by using the [vagrant-proxyconf](https://github.com/tmatilai/vagrant-proxyconf) plugin for Vagrant to configure both the default proxy as well as any internal exceptions.

Here's an example of a configuration file we'd include for those who use vagrant-proxyconf:
```
  if Vagrant.has_plugin?("vagrant-proxyconf")
    config.proxy.http     = ENV["http_proxy"]
    config.proxy.https    = ENV["https_proxy"]
    config.proxy.no_proxy = [
      "localhost", 
      "127.0.0.1", 
      (ENV["LATTICE_SYSTEM_IP"] || "192.168.11.11"), 
      (ENV["LATTICE_SYSTEM_DOMAIN"] || "192.168.11.11.xip.io"), 
      ".consul"
    ].join(',')
  end
```

We like this solution because it reduces the bug to a configuration change, rather than a more complex solution. However, this wouldn't work for Cloud-based deploys via Terraform. We feel that, with rare exceptions (OpenStack), most cloud installations of Lattice will have unfettered access to the Internet.

There are other routes we could take here, but this is our first suggestion on how to resolve the problem. Please let us know if this isn't satisfactory, or if there are other ways of approaching the problem that would be preferable.

--
  Marco Nicosia
  Product Manager
  Pivotal Software, Inc.

**Comment by menicosia** [comment]
Hi @dhilipkumars,

We've just released [Lattice v0.3.3](https://github.com/cloudfoundry-incubator/lattice/releases/tag/v0.3.3) which includes a fix for this bug [[#101547386](https://www.pivotaltracker.com/story/show/101547386)]. Please check it out and let us know what you think!

I'm going to close this issue for now, but as always, feel free to re-open or open a new issue if you think we've missed something!

--
Marco Nicosia
Product Manager
Pivotal Software, Inc.

**Comment by dhilipkumars** [comment]
Hi @menicosia Apologies for the delay again.   I confirm that now build-droplets command are working awesomely in our environment behind proxy.  

However i had use the `-e` option so that the container can download build packs.  (as already reccomended by you)
```
$time ltc build-droplet LAPP 'https://github.com/cloudfoundry/go-buildpack.git' -t "10m0s" -e http_proxy=http://proxy.com -e https_proxy=https://proxy.com -e no_proxy=192.168.11.11,xip.io,127.0.0.1,localhost
08/29 01:13:11.26 [BUILD|0] Successfully created container
08/29 01:13:11.75 [BUILD|0] Deleted http://192.168.11.11.xip.io:8444/blobs/LAPP/bits.zip.
08/29 01:13:11.76 [BUILD|0] Exit status 0
08/29 01:13:11.80 [BUILD|0] Exit status 0
08/29 01:14:19.76 [BUILD|0] -------> Buildpack version 1.6.0
08/29 01:14:19.81 [BUILD|0] -----> Checking Godeps/Godeps.json file.
08/29 01:15:19.75 [BUILD|0] -----> Installing go1.4.2... done
08/29 01:15:19.77 [BUILD|0] -----> Running: godep go install -tags cloudfoundry ./...
08/29 01:15:30.86 [BUILD|0] Exit status 0
08/29 01:15:30.94 [BUILD|0] Uploaded /tmp/droplet to http://192.168.11.11.xip.io:8444/blobs/LAPP/droplet.tgz.
08/29 01:15:30.94 [BUILD|0] Exit status 0
08/29 01:15:30.99 [BUILD|0] Uploaded /tmp/result.json to http://192.168.11.11.xip.io:8444/blobs/LAPP/result.json.
08/29 01:15:31.00 [BUILD|0] Exit status 0
Build completed

real    2m20.713s
user    0m0.180s
sys     0m0.042s


```
Thank you very much for the help.
